tests/integration/azure_blob/test_azure_blob.py

```diff
@@ -32,90 +34,91 @@
 }
 
 
-def azure_connection_configured():
-    return config["env"].get("AZURE_STORAGE_CONNECTION_STRING") is not None
-
-
 def prepare_env(auth_method):
-    # Remove any previously existing env_vars needed for authentication
+    if not config["env"].get("AZURE_CONTAINER"):
+        return False
+
     for k, env_vars in AUTH_METHODS_AND_REQUIRED_PARAMS.items():
         for env_var in env_vars:
             os.environ.pop(env_var, None)
-    # Then set the env_var needed to test a specific authentication method
-    env_vars = AUTH_METHODS_AND_REQUIRED_PARAMS.get(auth_method)
-    for env_var in env_vars:
-        os.environ[env_var] = config["env"].get(env_var)
 
-
-@pytest.mark.skipif(
-    not azure_connection_configured(),
-    reason="This is an integration test, add the needed environment variables in test-azure-blob.yml "
-    "to run it",
+    test_params = AUTH_METHODS_AND_REQUIRED_PARAMS.get(auth_method)
+    if not test_params:
+        return False
+
+    for env_var in test_params:
+        env_value = config["env"].get(env_var)
+        if not env_value:
+            return False
+        os.environ[env_var] = env_value
+
+    logger.info(f"Testing auth method {auth_method}")
+    return True
+
+
+pytestmark = pytest.mark.parametrize(
+    "auth_method",
+    [
+        pytest.param(
+            key,
+            marks=pytest.mark.skipif(
+                not prepare_env(key), reason=f"Auth method {key} not configured."
+            ),
+        )
+        for key in AUTH_METHODS_AND_REQUIRED_PARAMS
+    ],
 )
-def test_azure_blob():
-    for auth_method in AUTH_METHODS_AND_REQUIRED_PARAMS.keys():
 
-        prepare_env(auth_method)
 
-        blob_path = "az://" + config["env"].get("AZURE_CONTAINER")
-        blob_url = blob_path + "/" + blob_dir + "/" + blob_file
+def test_azure_blob(auth_method):
+    prepare_env(auth_method)
```

Comment:
When pytest will check whether to skip the test it will trigger this function, isn't that enough ? 
Why triggering it again ?

Reply:
`prepare_env` sets parameters for a specific auth method, and clears all other parameters. In the beginning pytest will iterate over all methods and leave the parameters in the state of the last auth method used. Therefore, in the beginning of each iteration we need to re-establish the parameters for that specific auth method.